mlir-dialect-experiments/exec-mlir-from-rust/Makefile

```diff
@@ -0,0 +1,10 @@
+all:
+	clang -c mlir-src/hello.c \
+		    -o mlir-target/hello.o && \
+	llvm-ar rc \
```

Comment:
This isn't good makefile style:

* Make knows how to build libraries, you just have to override `AR` if you want `llvm-ar` to be used.
* Make also knows how to build a `.o` from a `.c`, just override `CC` to `clang`
* This should be split into mutliple rules, one for building the `.o` one for creating the library.